testnet/stacks-node/src/tests/nakamoto_integrations.rs

```diff
@@ -5303,3 +5316,1287 @@ fn continue_tenure_extend() {
+            .get_stacks_blocks_processed();
+        if blocks_processed > blocks_processed_before {
+            break;
+        }
+        thread::sleep(Duration::from_millis(100));
+    }
+
+    let time0b_value = call_read_only(
+        &naka_conf,
+        &sender_addr,
+        contract0_name,
+        "get-time",
+        vec![&clarity::vm::Value::UInt(last_stacks_block_height)],
+    );
+    let time0b = time0b_value
+        .expect_optional()
+        .unwrap()
+        .unwrap()
+        .expect_u128()
+        .unwrap();
+    assert_eq!(
+        time0a, time0b,
+        "get-block-info? time should not have changed"
+    );
+
+    let time1b_value = call_read_only(
+        &naka_conf,
+        &sender_addr,
+        contract1_name,
+        "get-time",
+        vec![&clarity::vm::Value::UInt(last_stacks_block_height)],
+    );
+    let time1b = time1b_value
+        .expect_optional()
+        .unwrap()
+        .unwrap()
+        .expect_u128()
+        .unwrap();
+    assert_eq!(
+        time0b, time1b,
+        "Time from pre- and post-epoch 3.0 contracts should match"
+    );
+
+    let time3b_block_value = call_read_only(
+        &naka_conf,
+        &sender_addr,
+        contract3_name,
+        "get-block-time",
+        vec![&clarity::vm::Value::UInt(last_stacks_block_height)],
+    );
+    let time3b_block = time3b_block_value
+        .expect_optional()
+        .unwrap()
+        .unwrap()
+        .expect_u128()
+        .unwrap();
+
+    assert!(
+        time3b_block - time3a_block >= 1,
+        "get-stacks-block-info? time should have changed"
+    );
+
+    coord_channel
+        .lock()
+        .expect("Mutex poisoned")
+        .stop_chains_coordinator();
+    run_loop_stopper.store(false, Ordering::SeqCst);
+
+    run_loop_thread.join().unwrap();
+}
+
+fn assert_block_info(
+    tuple0: &BTreeMap<ClarityName, Value>,
+    miner: &clarity::vm::Value,
+    miner_spend: &clarity::vm::Value,
+) {
+    assert!(tuple0
+        .get("burnchain-header-hash")
+        .unwrap()
+        .clone()
+        .expect_optional()
+        .unwrap()
+        .is_some());
+    assert!(tuple0
+        .get("id-header-hash")
+        .unwrap()
+        .clone()
+        .expect_optional()
+        .unwrap()
+        .is_some());
+    assert!(tuple0
+        .get("header-hash")
+        .unwrap()
+        .clone()
+        .expect_optional()
+        .unwrap()
+        .is_some());
+    assert_eq!(
+        &tuple0
+            .get("miner-address")
+            .unwrap()
+            .clone()
+            .expect_optional()
+            .unwrap()
+            .unwrap(),
+        miner
+    );
+    assert!(tuple0
+        .get("time")
+        .unwrap()
+        .clone()
+        .expect_optional()
+        .unwrap()
+        .is_some());
+    assert!(tuple0
+        .get("vrf-seed")
+        .unwrap()
+        .clone()
+        .expect_optional()
+        .unwrap()
+        .is_some());
+    assert!(tuple0
```

Comment:
But presumably, you can query the _matured_ rewards too, right?  Is this tested somewhere?

Reply:
Ah, I see you have a separate test for this :+1:

Reply:
Yes, this is checked in the test that is too slow to add to the normal CI, `check_block_info_rewards`.